onnxscript/rewriter/_fusion_utils.py

```diff
@@ -26,11 +27,18 @@ def _check_shape(bindings: dict[str, Dim], val: ir.Value, shape: Sequence[str])
 def apply_fusion_rules(rules: pattern.RewriteRule | pattern.RewriteRuleSet) -> Callable:
     """
     Apply the given fusion rules to the model and return the number of fusions applied.
-    If debug is True, enable pattern matching tracer for debugging.
+
+    model: The input ONNX model represented as an `ir.Model`.
+    debug: If debug is True, enable pattern matching tracer for debugging.
+    apply_shape_inference: If True, apply shape inference after fusions.
     """
 
-    def apply_to(model: ir.Model, debug: bool = False) -> int:
+    def apply_to(
+        model: ir.Model, debug: bool = False, apply_shape_inference: bool = False
+    ) -> int:
         count = rules.apply_to_model(model)
+        if apply_shape_inference:
+            model = shape_inference.infer_shapes(model)
```

Comment:
This is not going to work, since shape-inference does not update model in-place. The returned model is ignored here. If your tests are working, maybe you don't need this feature?

Reply:
Yep, I thought it was in-place given https://github.com/microsoft/onnxscript/blob/b63ba43ca2eb64d3ee7186117ececd810ac7adaf/onnxscript/ir/passes/common/shape_inference.py#L53

Reply:
oh, interesting! Didn't realize that it merges it back to original model. In that case, I suggest dropping the assignment to model, and writing it as `shape_inference.infer_shapes(model)`.

Reply:
This implies that `fuse_xformers` itself becomes an in-place fusion. Which is good for a performance perspective. But we should probably update the signature (back) to clarify this or at least document it.